### PR TITLE
Worker fixes

### PR DESCRIPTION
**Pull request by danixeee**
- Plainly provider wrapped in try-catch
- Better logging
- Font installer fixes

### CODE REVIEW

File: packages/nexrender-action-fonts-plainly/src/fontsOsx.js [hunk 1]
```diff
@@ -4,19 +4,36 @@ const { copyFile, removeFile } = require('./utils');
...
     fonts.map(async (f) => {
+      const fontName = basename(f);
       const fontSrc = join(workDir, f);
-      await copyFile(fontSrc, join(FONTS_DIRECTORY, basename(f)));
+
+      try {
+        await copyFile(fontSrc, join(FONTS_DIRECTORY, fontName));
+
+        settings.logger.log(`Font ${fontName} [${fontSrc}] installed.`);
+      } catch (e) {
```
Comment (ivansenic) [inline packages/nexrender-action-fonts-plainly/src/fontsOsx.js:17]: it's funny that in my review I asked about this, cccccc :angry:

File: packages/nexrender-action-fonts-plainly/src/fontsOsx.js [hunk 2]
```diff
@@ -4,19 +4,36 @@ const { copyFile, removeFile } = require('./utils');
 
 const FONTS_DIRECTORY = join(homedir(), "Library", "Fonts");
 
-const install = async (workDir, fonts) => {
+const install = async (workDir, fonts, settings) => {
   return Promise.all(
     fonts.map(async (f) => {
+      const fontName = basename(f);
```
Comment (ivansenic) [inline packages/nexrender-action-fonts-plainly/src/fontsOsx.js:10]: why is not everything in the try, why basename and src outside?

File: packages/nexrender-provider-plainly/src/index.js [hunk 3]
```diff
@@ -33,32 +33,37 @@ const getTemplateDir = async (templateBucket) => {
...
+      const localAepFilePath = path.join(localTemplateDir, path.basename(src));
+      await fs.promises.copyFile(localAepFilePath, dest);
+      // Create footage shortcut (instead of fixing project paths...)
+      const localFootagePath = path.join(localTemplateDir, '(Footage)');
+      const destFootagePath = path.join(path.dirname(dest), '(Footage)');
+      await fs.promises.symlink(localFootagePath, destFootagePath, 'junction');
+      // Replace job template src
+      job.template.src = localAepFilePath;
+    } catch (e) {
+      log(e.message);
```
Comment (ivansenic) [inline packages/nexrender-provider-plainly/src/index.js:65]: can we also log here our own message in addition, so we can find it?

File: packages/nexrender-provider-plainly/src/index.js [hunk 4]
```diff
@@ -33,32 +33,37 @@ const getTemplateDir = async (templateBucket) => {
 
 const download = async (job, settings, src, dest, params, type) => {
     const log = fromLogger(settings.logger.log);
-    const download = (s, d) => gsProvider.download(job, settings, s, d, params, type);
 
-    src = src.replace(`${PLAINLY_PROTOCOL}//`, '');
-    const templateRootDir = path.dirname(src);
-    const localTemplateDir = await getTemplateDir(templateRootDir);
+    try {
```
Comment (ivansenic) [inline packages/nexrender-provider-plainly/src/index.js:37]: ok all these changes are try/catch wrap?

**Review by ivansenic — APPROVED** [review]
Check comments, let's merge and let's redeploy asap..

File: packages/nexrender-provider-plainly/src/index.js (hunk 4 above)
Comment (danixeee) [reply packages/nexrender-provider-plainly/src/index.js:37]: yup

File: packages/nexrender-provider-plainly/src/index.js (hunk 3 above)
Comment (danixeee) [reply packages/nexrender-provider-plainly/src/index.js:65]: it's already prefixed, look here https://github.com/plainly-videos/nexrender/blob/master/packages/nexrender-provider-plainly/src/index.js#L14

**Pull request merged**